### PR TITLE
Attending or not logic

### PR DESCRIPTION
**Pull request by castaneda-ed**
So I uncommented al the auth related code, also added a path to the register/login component, and  did some changes to eventlist and evencard files for the attending/not attending logic to work, The most important thing is that I installed jwt-decode, to decode users token to get the id and use it for the buttons going/not going functionality, I used the request body id to add or delete a user from the backend, but that's only usable inside the request updateEvent itself, I needed some sort of state to use the users id for my buttons functionality, I decoded the id from the token for that in the EventComponent file, so that I did not change anything in the authSlice cause that means a lot more complexity.

### CODE REVIEW

File: src/components/events/EventCard.jsx [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
+      //Extracts the id of the the decoded users token
+      const { id: userId } = jwtDecode(token);
+      const isAttending = event.attendingUsers?.some(
+        (user) => user.id === userId
+      );
+      setAttending(isAttending);
+    }
+  }, [token, event.attendingUsers]);
+
+  const [updateEvent] = useUpdateEventMutation();
```
Comment (castaneda-ed) [inline src/components/events/EventCard.jsx]: Here I used use effect to only run the code when  token (user) changes or the attneding user change.  Used jwt to decode the id from token.

File: src/components/events/EventCard.jsx [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+    try {
+      await updateEvent({
+        id: event.id,
+        attending: newStatus,
+      });
+      setAttending(newStatus);
+    } catch (error) {
+      console.error("Error updating attendance:", error);
+    }
+  };
```
Comment (castaneda-ed) [inline src/components/events/EventCard.jsx:45]: The req.user.id would only work in this async function and I needed it on my state, so I used useEffect and decode. :)

**Comment by beefchief-code** [comment]
Screenshot 2024-10-31 at 9 58 29 PM

keep getting this error when i try and test the site. looks like everything is correct from the jwt-decode documentation so might be something on my end but we can take a look in the morning

**Comment by castaneda-ed** [comment]
I just pushed here the fixed jwt-decode, I had installed a past version I just installed the latest version and fixed the import.

**Review by rmartinezusa — APPROVED** [review]

File: src/components/events/EventCard.jsx
Comment (beefchief-code) [inline src/components/events/EventCard.jsx:1]: jwt-decode is for backend. proper way to get id of user is /users/me route that bertha is writing, will be up soon (11/1). pull lester in for more info :)

Comment (beefchief-code) [reply src/components/events/EventCard.jsx:1]: should be a useQuery from berthas incoming code

Comment (castaneda-ed) [reply src/components/events/EventCard.jsx:1]: O I see, we still needed the users routes to get that info. Ok Ill wait for that and Update my code

**Pull request closed without merging**